sopel/modules/dice.py

```diff
@@ -218,12 +218,20 @@ def _get_pretty_str(dice):
     eval_str = arg_str % (tuple(map(_get_eval_str, dice)))
     pretty_str = arg_str % (tuple(map(_get_pretty_str, dice)))
 
-    # Showing the actual error will hopefully give a better hint of what is
-    # wrong with the syntax than a generic error message.
     try:
         result = eval_equation(eval_str)
-    except Exception as e:
-        bot.reply("SyntaxError, eval(%s), %s" % (eval_str, e))
+    except TypeError:
+        bot.reply("The type of this equation is, apparently, not a string. " +
+            "How did you do that, anyway?")
+    except ValueError:
+        # As it seems that ValueError is raised if the resulting equation would
+        # be too big, give a semi-serious answer to reflect on this.
+        bot.reply("You roll %s: %s = very big" % (
+            trigger.group(2), pretty_str))
+        return
+    except (SyntaxError, eval_equation.Error):
```

Comment:
This reference to `sopel.tools.calculation.ExpressionEvaluator.Error` is weird done this way, but it's not really possible to do it more conventionally due to the nested classes. Trust me, I tried! :joy_cat:

Frustratingly, doing e.g. `from sopel.tools.calculation.ExpressionEvaluator import Error as ExpressionEvaluationError` doesn't work. Nor does `from sopel.tools.calculation import ExpressionEvaluator.Error as ExpressionEvaluationError`. That's invalid syntax.

I'm not up for refactoring `sopel.tools.calculation` any time soon, just to "fix" this one `except` statement to look more conventional, so the module and this PR will do as-is. Maybe a future Sopel release (major version) will eventually make it so the calculation tools actually provides a proper `ExpressionEvaluationError` that can be imported on its own for catching exceptions… but that's definitely not happening in the point release that will include this PR.